neuralpp/symbolic/context_simplifier.py

```diff
@@ -48,7 +49,9 @@ def _simplify_pass(expression: Expression, context: Z3SolverExpression) -> Expre
             result = expression
 
         # replace boolean expressions
-        all_boolean_subexpressions = _collect_subset_expressions(result, lambda expr: expr.type == bool)
+        all_boolean_subexpressions = _collect_subset_expressions(
+            result,
+            lambda expr: expr.type == bool and not isinstance(expr, QuantifierExpression))
```

Comment:
Simplifier should not look inside quantifier because of scope issue: indexed expression should not be visible outside the quantifier.